scripts/skill_auditor/test_metrics_extractor.py

```
@@ -0,0 +1,513 @@
+    """Test that UnicodeDecodeError on SKILL.md is handled gracefully."""
+    with tempfile.TemporaryDirectory() as tmp:
+        tmp_path = Path(tmp)
+        skill_md = tmp_path / "SKILL.md"
+
+        # Write invalid UTF-8 bytes
+        skill_md.write_bytes(b"\xff\xfe Invalid UTF-8 \x80\x81")
+
+        try:
+            extract_skill_metrics(tmp_path)
+            # If we get here, the test fails - no exception was raised
+            assert False, "Expected ValueError to be raised but it was not"
```

Ruff (0.14.5)

Assertion always fails, replace with `pytest.fail()`

(PT015)